celery/utils/serialization.py

```diff
@@ -137,25 +145,25 @@ def __str__(self):
         return self.text
 
     @classmethod
-    def from_exception(cls, exc):
+    def from_exception(cls, exc, dumps=pickle.dumps):
         return cls(exc.__class__.__module__,
                    exc.__class__.__name__,
                    getattr(exc, 'args', []),
                    safe_repr(exc))
 
 
-def get_pickleable_exception(exc):
+def get_pickleable_exception(exc, loads=pickle.loads, dumps=pickle.dumps):
     """Make sure exception is pickleable."""
     try:
-        pickle.loads(pickle.dumps(exc))
```

Comment:
I am not sure I understand the changes in this function. Could you please explain the logic here?

Reply:
That's for reviewing @georgepsarakis. This got rushed a bit -- I'll clean it up.